### ISSUE TITLE
Invalid field prefix 'toolset' in filter

### ISSUE THREAD

**Issue by JohannesMP** [premake/premake-core#257]
According to the <a href=https://github.com/premake/premake-core/wiki/filter>filter documentation</a> the `toolset` field prefix **is** supported in filters.

However when my `premake5.lua` contains the following:

```
-- compiling for c++14
filter { "toolset:clang" }
  buildoptions { "-std=c++14" }
```

I get an error stating `Invalid field prefix 'toolset'` for the line with the filter.

Is this a bug or is the documentation outdated?

**Comment by tritao** [comment]
Hey, I also hit this and tried to fix it in the referenced pull but it's still not working.

**Comment by jeremyong** [comment]
If you look at the code for various toolsets like gcc.lua or clang.lua, you can see that the flags C++11 and C++14 are automatically mapped in the correct place. If you replace what you have with

```lua
flags { "C++14" }
```

things should work.

**Comment by JohannesMP** [comment]
I appreciate the attempt to help, but that was just an example; not what I actually was hoping to use it for in my actual projects. The point is that a filter which is stated in the documentation to work, doesn't.

**Comment by starkos** [comment]
@tritao has tried to fix this; feedback on PR #265 would be appreciated.

**Comment by samsinsane** [comment]
Closing this since #265 was merged.
